embedded-graphics/src/geometry/point.rs

```diff
@@ -0,0 +1,368 @@
+use crate::geometry::Size;
+use core::ops::{Add, AddAssign, Index, Neg, Sub, SubAssign};
+
+/// 2D point.
+///
+/// A point can be used to define the position of a graphics object. For example, a [`Rectangle`]
+/// may be defined that has its top left corner at `(-1, -2)`. To specify the size of an object
+/// [`Size`] should be used instead.
+///
+/// [Nalgebra] support can be enabled with the `nalgebra_support` feature. This implements
+/// `From<Vector2<N>>` and `From<&Vector2<N>>` where `N` is `Scalar + Into<i32>`. This allows use
+/// of Nalgebra's [`Vector2`] with embedded-graphics where `i8`, `i16`, `i32`, `u16` or `u8` is used
+/// for value storage.
+///
+/// # Examples
+///
+/// ## Create a `Point` from two integers
+///
+/// ```rust
+/// use embedded_graphics::geometry::{point, Point};
+///
+/// // Create a coord using the `new` constructor method
+/// let c1 = Point::new(10, 20);
+///
+/// // Create a coord using the handy `point` function
+/// let c2 = point(10, 20);
+///
+/// assert_eq!(c1, c2);
+/// ```
+///
+/// ## Create a `Point` from a Nalgebra `Vector2`
+///
+/// _Be sure to enable the `nalgebra_support` feature to get [Nalgebra] integration._
+///
+/// ```rust
+/// # #[cfg(feature = "nalgebra_support")] {
+/// use embedded_graphics::geometry::Point;
+/// use nalgebra::Vector2;
+///
+/// let n_coord = Vector2::new(10i32, 20);
+///
+/// assert_eq!(Point::from(n_coord), Point::new(10, 20));
+/// # }
+/// ```
+///
+/// ## Convert a `Vector2<u8>` into a `Point`
+///
+/// _Be sure to enable the `nalgebra_support` feature to get [Nalgebra] integration._
+///
+/// Smaller unsigned types that can be converted to `i32` are also supported in conversions.
+///
+/// ```rust
+/// # #[cfg(feature = "nalgebra_support")] {
+/// use embedded_graphics::geometry::Point;
+/// use nalgebra::Vector2;
+
+/// let n_coord = Vector2::new(10u8, 20);
+///
+/// assert_eq!(Point::from(n_coord), Point::new(10, 20));
+/// # }
+/// ```
+///
+/// [`Size`]: struct.Size.html
+/// [`Rectangle`]: ../primitives/rectangle/struct.Rectangle.html
+/// [`Vector2<N>`]: https://docs.rs/nalgebra/0.18.0/nalgebra/base/type.Vector2.html
+/// [`Vector2`]: https://docs.rs/nalgebra/0.18.0/nalgebra/base/type.Vector2.html
+/// [Nalgebra]: https://docs.rs/nalgebra
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct Point {
+    /// The x coordinate.
+    pub x: i32,
+
+    /// The y coordinate.
+    pub y: i32,
+}
+
+impl Point {
+    /// Create a new coordinate with X and Y values
+    pub const fn new(x: i32, y: i32) -> Self {
+        Point { x, y }
+    }
+
+    /// Creates a point with x and y equal to zero.
```

Comment:
Capitalisation consistent with above doc comment

```suggestion
    /// Creates a point with X and Y equal to zero.
```